src/agent/onefuzz-telemetry/src/lib.rs

```diff
@@ -450,7 +450,10 @@ mod global {
 }
 
 const REDACTED: &str = "Redacted";
-pub fn set_appinsights_clients(
+// This function doesn't do anything async, but TelemetryClient::new must be invoked
```

Comment:
Should this be also be enforced upstream by making `TelemetryClient::new()` an `async` method in the `audit-fixes` fork?

Reply:
Taking this to the Rustaceans channel for discussion 🙂

Reply:
FWIW: I think this PR is good-to-go regardless. If you implement this idea, it'd force the `async`-ness change anyway. Though it _would_ break the build, which prompted me to add one more suggestion re our the dependency spec for your fork.

Reply:
@ranweiler Yes I am inclined to leave it as-is for now since that would be a source-breaking change whereas the current PR should (ha!) not affect downstream users at all.